plugins/color_panel/lib/color_panel.dart

```diff
@@ -0,0 +1,121 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+import 'dart:async';
+import 'dart:ui';
+
+import 'package:flutter/services.dart';
+
+const _kColorPanelChannel = 'flutter/colorpanel';
+const _kShowColorPanelMethod = 'ColorPanel.Show';
+const _kHideColorPanelMethod = 'ColorPanel.Hide';
+const _kColorPanelColorSelectedCallback = 'ColorPanel.ColorSelectedCallback';
+const _kColorPanelClosedCallback = 'ColorPanel.ClosedCallback';
+
+const _kRedKey = 'red';
+const _kGreenKey = 'green';
+const _kBlueKey = 'blue';
+
+const _platformChannel =
+    const MethodChannel(_kColorPanelChannel, const JSONMethodCodec());
+
+/// A callback to pass to [ColorPanel] to receive user-selected colors.
+typedef ColorPanelCallback = void Function(Color color);
+
+/// Provides access to an OS-provided color chooser.
+///
+/// This class is a singleton, since the OS may not allow multiple color panels
+/// simultaneously.
+class ColorPanel {
+  ColorPanelCallback _callback;
+
+  /// Private constructor.
+  ColorPanel._() {
+    _platformChannel.setMethodCallHandler(_wrappedColorPanelCallback);
+  }
+
+  /// The static instance of the panel.
+  static ColorPanel instance = new ColorPanel._();
+
+  /// Whether or not the panel is showing.
+  bool get showing => _callback != null;
+
+  /// Shows the color panel.
+  ///
+  /// [callback] will be called when the user selects a color; it can be called
+  /// an number of times depending on the interaction model of the native
+  /// panel.
+  ///
+  /// It is an error to call [show] if the panel is already showing.
+  void show(ColorPanelCallback callback) {
+    try {
+      if (!showing) {
+        _callback = callback;
+        _platformChannel.invokeMethod(_kShowColorPanelMethod);
+      } else {
+        throw new StateError('Color panel is already shown');
+      }
+    } on PlatformException catch (e) {
+      print('PLATFORM ERROR: ${e.message}');
+    }
+  }
+
+  /// Hides the color panel.
+  ///
+  /// It is an error to call [hide] if the panel is not showing. Since
+  /// the panel can be closed by the user, you should always check
+  /// [showing] before calling [hide].
+  void hide() {
+    try {
+      if (showing) {
+        _platformChannel.invokeMethod(_kHideColorPanelMethod);
+        _callback = null;
+      } else {
+        throw new StateError('Color panel is already hidden');
+      }
+    } on PlatformException catch (e) {
+      print('PLATFORM ERROR: ${e.message}');
+    }
+  }
+
+  /// Given a color component value from 0-1, converts it to an int
+  /// from 0-255.
+  int _colorComponentFloatToInt(num value) {
+    const colorMax = 255;
+    return (value * colorMax).toInt();
+  }
+
+  /// Mediates between the platform channel callback and the client callback.
+  Future<Null> _wrappedColorPanelCallback(MethodCall methodCall) {
```

Comment:
Does this need to be `Future<Null>`?  It looks like this function is synchronous.

Reply:
If we don't return an explicit `new Future` then I think we should add the `async` keyword, as that'll wrap the return value in a Future and prevent potential weird things from happening.

Reply:
Done